Оптимизируй это (часть 1)

imbrickiy opened the pull request:
предложения, находящиеся рядом. Добавлен скролл при выборе предложения. Правлены маршруты.

---
:mortar_board: [Оптимизируй это (часть 1)](https://up.htmlacademy.ru/react/12/user/2171409/tasks/15)

imbrickiy committed:
Небольшая оптимизация с хуком useMemo. Добавлены
предложения, находящиеся рядом. Добавлен скролл при выборе предложения.
Правлены маршруты.

saimon395 commented on project/src/pages/main/main.tsx:
```diff
@@ -15,8 +15,9 @@ function MainScreen(): JSX.Element {
   const selectedCity = useAppSelector(({ city }) => city);
   const currentOffers = useAppSelector(({ offers }) => offers.filter((offer) => offer.city.name === selectedCity.name));
+  const currentedOffers = useMemo(() => currentOffers, [currentOffers]);
   const activeSortType = useAppSelector(({ sortOption }) => sortOption);
-  const sortedOffers = sortOffers(currentOffers, activeSortType);
+  const sortedOffers = sortOffers(currentedOffers, activeSortType);
```
этого маловато, сделай то бы хотя бы весь список офферов не перерисовывался при наведении на оффер, нужно будет ещё компонент OffersListComponent замемоизировать, и сортированный список офферов

imbrickiy committed:
Весь список офферов не перерисовывается при
наведении на оффер. компонент OffersListComponent и сортированный список офферов
- замемоизированы.

imbrickiy committed:
Рефакторинг компонента

saimon395 commented on project/src/components/offers-list/offers-list.tsx:
```diff
@@ -16,6 +16,7 @@ function OffersListComponent(props: OffersListComponentProps) {
     </div>
   );
-}
+});
+OffersListComponent.displayName = 'OffersListComponent';
```
не надо такого, пиши компонент как обычно, просто экспорт сделай такой 
`export default memo(OffersListComponent);`

saimon395 commented on project/src/pages/main/main.tsx:
```diff
@@ -15,9 +15,10 @@ function MainScreen(): JSX.Element {
   const currentOffers = useAppSelector(({ offers }) => offers.filter((offer) => offer.city.name === selectedCity.name));
+  const currentedOffers = useMemo(() => currentOffers, [currentOffers]);
   const activeSortType = useAppSelector(({ sortOption }) => sortOption);
-  const sortedOffers = sortOffers(currentOffers, activeSortType);
-
+  const sortedOffers = (sortOffers(currentedOffers, activeSortType));
+  const memoizedSortedOffers = useMemo(() => sortedOffers, [sortedOffers]);
```
как то очень сложно, к тому же не работает, тебе здесь нужно что то такое
```
  const offers = useAppSelector(({ offers }) => offers);

  const activeSortType = useAppSelector(({ sortOption }) => sortOption);

  const currentOffers = useMemo(() => {
    const currentOffers = offers.filter((offer) => offer.city.name === selectedCity.name);
    return sortOffers(currentOffers, activeSortType);
  }, [offers, activeSortType]);
```
